python/tchannel/tornado/message_factory.py

```diff
@@ -36,6 +41,156 @@ def __init__(self):
+        """
+        response.flags = FlagsType.none if is_completed else FlagsType.fragment
+
+        # TODO decide what need to pass from request
+        if response.state == StreamState.init:
+            message = CallResponseMessage(
+                flags=response.flags,
+                # code=response.code,
+                # tracing=response.tracing,
+                # headers=response.headers,
+                # checksum=response.checksum
+                args=args
+            )
+            response.state = (StreamState.completed if is_completed
+                              else StreamState.streaming)
+        elif response.state == StreamState.streaming:
+            message = CallResponseContinueMessage(
+                flags=response.flags,
+                # checksum=response.checksum
+                args=args
+            )
+            response.state = (StreamState.completed if is_completed
+                              else StreamState.streaming)
+        else:
+            raise StreamingException("response state Error")
+
+        return message
+
+    def build_raw_message(self, context, args, is_completed=False):
+        if isinstance(context, Request):
+            return self.build_raw_request_message(context, args, is_completed)
+        elif isinstance(context, Response):
+            return self.build_raw_response_message(context, args, is_completed)
+        else:
+            raise StreamingException("context object type error")
+
+    def build_request(self, message, message_id=None):
+        """Build request object from protocol level message info
+
+        It is allowed to take incompleted CallRequestMessage. Therefore the
+        created request may not contain whole three arguments.
+
+        :param message: CallRequestMessage
+        :param message_id: integer of message id
+        :return: request object
+        """
+        args = []
+        for arg in message.args:
+            args.append(InMemArgStream(arg, auto_close=False))
+
+        # TODO decide what to pass to Request from message
+        req = Request(
+            flags=message.flags,
+            # ttl=message.ttl,
+            # tracing=message.tracing,
+            service=message.service,
+            # headers=messaget.headers,
+            # checksum=message.checksum
+            argstreams=args,
+            id=message_id
+        )
+        return req
+
+    def build_response(self, message, message_id=None):
+        """Build response object from protocol level message info
+
+        It is allowed to take incompleted CallResponseMessage. Therefore the
+        created request may not contain whole three arguments.
+
+        :param message: CallResponseMessage
+        :param message_id: integer of message id
+        :return: response object
+        """
+
+        args = []
+        for arg in message.args:
+            args.append(InMemArgStream(arg, auto_close=False))
```

Comment:
Because on one hand we are receiving fragments from socket, on the other hand user may start to consume. If we set auto_close to true, argsream may closed automatically. Therefore set it false here, we will close stream when we receive all fragments